episodes/vaccine-comparisons.Rmd

```diff
@@ -0,0 +1,968 @@
+
+close_schools_early_20 <- intervention(
+  name = "School closure",
+  type = "contacts",
+  time_begin = early_start,
+  time_end = early_start + duration_20,
+  reduction = matrix(c(0.5, 0.01, 0.01))
+)
+
+close_schools_early_50 <- intervention(
+  name = "School closure",
+  type = "contacts",
+  time_begin = early_start,
+  time_end = early_start + duration_50,
+  reduction = matrix(c(0.5, 0.01, 0.01))
+)
+
+output_npi_early_vacc_late_20 <- model_default(
+  population = uk_population,
+  transmission_rate = transmission_rate,
+  infectiousness_rate = infectiousness_rate,
+  recovery_rate = recovery_rate,
+  vaccination = vacc_late,
+  intervention = list(contacts = close_schools_early_20),
+  time_end = 300, increment = 1.0
+)
+
+output_npi_early_vacc_late_50 <- model_default(
+  population = uk_population,
+  transmission_rate = transmission_rate,
+  infectiousness_rate = infectiousness_rate,
+  recovery_rate = recovery_rate,
+  vaccination = vacc_late,
+  intervention = list(contacts = close_schools_early_50),
+  time_end = 300, increment = 1.0
+)
+
+output_npi_early_vacc_late_20$intervention_type <- "20 days"
+output_npi_early_vacc_late_50$intervention_type <- "50 days"
+output_npi_early_vacc_late$intervention_type <- "100 days"
+
+output_npis <- rbind(output_npi_early_vacc_late_20,
+                     output_npi_early_vacc_late_50,
+                     output_npi_early_vacc_late)
+
+output_npis %>%
+  filter(compartment == "infectious") %>%
+  ggplot() +
+  aes(
+    x = time,
+    y = value,
+    color = intervention_type,
+    linetype = intervention_type
+  ) +
+  stat_summary(
+    fun = "sum",
+    geom = "line",
+    linewidth = 1
+  ) +
+  scale_y_continuous(
+    labels = scales::comma
+  ) +
+  geom_vline(
+    xintercept = c(
+      early_start,
+      early_start + duration
+    ),
+    linetype = 2
+  ) +
+  geom_vline(
+    xintercept = c(
+      late_start,
+      late_start + duration
+    ),
+    linetype = 3
+  ) +
+  theme_bw() +
+  labs(
+    x = "Simulation time (days)",
+    y = "Individuals"
+  )
+```
+
+
+
+:::::::::::::::::::::::::::::::::
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+## Summary
+
+Using model generated disease trajectories we have investigated the potential impact of different vaccination strategies. When infection has group specific risk, targeted vaccination programs may be more effective in reducing epidemic peak. NPIs can be utilised alongside vaccine roll outs to control epidemic peaks when vaccines may not be readily available. 
+
+In this tutorial we have investigated the potential effect of infection blocking vaccines, but we can use models to explore other effects of vaccines including imperfect vaccines. The disease trajectories generated in this tutorial can be used to calculate measures such as DALYs averted as part of health economic analyses.
+
+
+::::::::::::::::::::::::::::::::::::: keypoints 
+
+- Herd immunity is a indirect effect of vaccination programs
```

Comment:
Probably beyond scope of this episode to model, but could get them to think about the impact vaccination would have if it came very late in the epidemic (i.e. after post-infection immunity had accumulated) – this happened in 2009, for example: https://pubmed.ncbi.nlm.nih.gov/20096762/